demo/api/src/app.module.ts

```diff
@@ -106,6 +107,7 @@ export class AppModule {
                     Documents: [Page, Link, PredefinedPage],
                     Scope: PageTreeNodeScope,
                     reservedPaths: ["/events"],
+                    sitePreviewSecret: randomBytes(32).toString("hex"),
```

Comment:
Shouldn't this be an environment variable for local development? Suppose I'm in the site preview and make a change to some API code (e.g., add a missing field to transformToPlain), the API restarts, and the site preview doesn't work anymore, if I understand it correctly?

Reply:
No. The validation only occurs when entering the preview mode. This is triggered when opening the preview but it is always preceded by the creation of the hash. This means the restart of the API has to take place between the two request, but this is a very small time frame and therefor neglectable.

We only have to make sure that the secret is the same between more than one instances of the API which is not the case in local development.